### PR TITLE
Switch to QE workflow and run sanity test on 4.14+ rollback

### PR DESCRIPTION
**Pull request by shellyyang1989**
The change switches the `oldest-supported-z-rollback` in 4.14 to 4.14.0 and `oldest-supported-z-rollback` in 4.16 to 4.16.0-ec.0. Also it switches the test to QE workflow.

### CODE REVIEW

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-aws-ovn-upgrade-rollback-oldest-supported periodic-ci-openshift-release-master-nightly-4.16-aws-ovn-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-aws-ovn-upgrade-rollback-oldest-supported periodic-ci-openshift-release-master-nightly-4.16-aws-ovn-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
@jianlinliu @jiajliu @JianLi-RH @dis016 for review. Thank you.

**Comment by jianlinliu** [comment]
> Also it switches the test to QE workflow.

Are this decision already synced between QE and Dev?

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.16.yaml [hunk 1]
```diff
@@ -585,18 +584,32 @@ tests:
...
+        name: release:oldest-supported-z-rollback
+      - env: RELEASE_IMAGE_LATEST
+        name: release:latest
+      from: cli
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    - ref: cucushift-chainupgrade-toimage
+    - chain: openshift-upgrade-qe-sanity
```
Comment (jiajliu) [inline ci-operator/config/openshift/release/openshift-release-master__nightly-4.16.yaml:610]: Just want to confirm is this sanity test what dev expected sanity test? There is not e2e test in `openshift-upgrade-qe-sanity`. From original job name, it seems e2e needed.

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml [hunk 2]
```diff
@@ -98,7 +98,7 @@ releases:
   oldest-supported-z-rollback:
     release:
       channel: candidate
-      version: 4.14.0-rc.0
+      version: 4.14.0
```
[the remark this answers is not in the document]
Comment (shellyyang1989) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml:101]: Right. Should move to 4.15.0

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-aws-ovn-upgrade-rollback-oldest-supported periodic-ci-openshift-release-master-nightly-4.16-aws-ovn-upgrade-rollback-oldest-supported periodic-ci-openshift-release-master-nightly-4.15-aws-ovn-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
/pj-rehearse periodic-ci-openshift-release-master-nightly-4.14-aws-ovn-upgrade-rollback-oldest-supported periodic-ci-openshift-release-master-nightly-4.16-aws-ovn-upgrade-rollback-oldest-supported periodic-ci-openshift-release-master-nightly-4.15-aws-ovn-upgrade-rollback-oldest-supported

**Comment by shellyyang1989** [comment]
4.15 [test](https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_release/48534/rehearse-48534-periodic-ci-openshift-release-master-nightly-4.15-aws-ovn-upgrade-rollback-oldest-supported/1764487172808773632) passed.

**Comment by shellyyang1989** [comment]
The 4.14 and 4.16 failures might be product issues or flake instead of CI issue. So

/pj-rehearse ack

File: ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml [hunk 3]
```diff
@@ -547,18 +547,31 @@ tests:
...
+      - env: RELEASE_IMAGE_LATEST
+        name: release:latest
+      from: cli
+      resources:
+        requests:
+          cpu: 100m
+          memory: 200Mi
+    - ref: cucushift-chainupgrade-toimage
+    - chain: openshift-upgrade-qe-sanity
+    workflow: cucushift-installer-rehearse-aws-ipi-ovn
```
[the remark this answers is not in the document]
Comment (shellyyang1989) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml:574]: Yes, currently, the testing involves running the upgrade-conformance test suite. However, the proposed change conducts a sanity test, which includes checks for MCP , cluster operators, machines, and pods. This adjustment was recommended by the OTA dev team during the sync call. cc @LalatenduMohanty

[the remark this answers is not in the document]
Comment (shellyyang1989) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml:574]: Hi @dgoodwin are there any more concern about it?

[the remark this answers is not in the document]
Comment (dgoodwin) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml:574]: I was concerned about a job suddenly changing all of it's tests, which could cause some problems in tooling, however it looks like the jobs actually have been renamed, via dropping the e2e now, so this concern ok.

I was wondering why the change in suite was made, and concerned that the job is losing a lot of signal. Those thousands of tests removed do a lot of interesting things that help verify a cluster was healthy. I will assume this is known and everyone agrees we don't want or need that signal on these rollback jobs.

[the remark this answers is not in the document]
Comment (shellyyang1989) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml:574]: I appreciate your concern. After extensive testing, we found that the conformance test suite was not stable in the rollback testing. Almost all of the failures detected in the conformance test suite were not directly related to the rollback process but rather represented false signals. Additionally, given that older versions typically have fewer bug fixes compared to newer versions, encountering some failures in older versions is not uncommon. To address these issues and ensure more accurate testing results, we made the decision to transition to a sanity test suite, which better aligns with the specific goals and requirements of rollback testing.

[the remark this answers is not in the document]
Comment (LalatenduMohanty) [reply ci-operator/config/openshift/release/openshift-release-master__nightly-4.14.yaml:574]: @shellyyang1989 is right. Also the expectation from rollback is not same as a typical upgrade. During the rollback we do not care much about the alerts or some noise (operator going to degraded) as long as post rollback the cluster is healthy.

**Comment by shellyyang1989** [comment]
Refers [OTA-1071](https://issues.redhat.com/browse/OTA-1071)

**Pull request merged**